src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -218,11 +217,19 @@ private set
                 ArgumentNullException.ThrowIfNull(value, nameof(FullName));
 
                 _storedEntryNameBytes = ZipHelper.GetEncodedTruncatedBytesFromString(
-                    value, _archive.EntryNameAndCommentEncoding, 0 /* No truncation */, out bool hasUnicodeEntryName);
+                    value, _archive.EntryNameAndCommentEncoding, 0 /* No truncation */, out bool isUTF8);
 
-                _hasUnicodeEntryNameOrComment |= hasUnicodeEntryName;
                 _storedEntryName = value;
 
+                if (isUTF8)
```

Comment:
I am not sure if I understand. When Utf8 is detected, we enable the Unicode flag?

Reply:
Yes. Maybe the name of the flag should be `utf8FileNameAndComment`.

Per the spec: https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT

```
 Bit 11: Language encoding flag (EFS).  If this bit is set,
                the filename and comment fields for this file
                MUST be encoded using UTF-8. (see APPENDIX D)
```

Reply:
> Yes. Maybe the name of the flag should be utf8FileNameAndComment to be very clear.

That would definitely help ;) thanks for the pointer to the docs!